helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/AvroSchemaUtil.java

```diff
@@ -529,6 +529,35 @@ public static Schema resolveUnionBranchOf(Object value, Schema union) {
     throw new IllegalStateException(sb.toString());
   }
 
+  /**
+   * checks if the value for a given schema can possibly contain
+   * strings. this is important when dealing with things like
+   * Utf8 vs java.lang.Strings
+   * @param schema a schema
+   * @return true if value under schema could possibly involve strings
+   */
+  public static boolean schemaContainsString(Schema schema) {
+    if (schema == null) {
+      return false;
+    }
+    boolean hasString = false;
+    switch (schema.getType()) {
```

Comment:
Curious if we should handle records here? If not, let's include it in the description/name given that this is a public util

Reply:
this doesnt traverse into records. updated the javadoc

Reply:
also it would hurt its intended usage if it traversed into records. when setting fields we do not need to convert records (any record field strings would have been taken care of before)